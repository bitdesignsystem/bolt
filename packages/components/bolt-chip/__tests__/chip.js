import {
  isConnected,
  render,
  renderString,
  stopServer,
  html,
} from '../../../testing/testing-helpers';

describe('<bolt-chip> Component', () => {
  afterAll(async () => {
    await stopServer();
  }, 100);

  test('basic usage with attributes', async () => {
    const results = await render('@bolt-components-chip/chip.twig', {
      attributes: {
        'data-attr': 'some attribute',
      },
      text: 'This is a chip',
    });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });

  test('with link', async () => {
    const results = await render('@bolt-components-chip/chip.twig', {
      text: 'This is a chip with link',
      url: '!#',
    });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });
});
