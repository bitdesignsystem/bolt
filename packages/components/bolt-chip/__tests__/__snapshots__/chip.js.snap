// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<bolt-chip> Component basic usage with attributes 1`] = `
<bolt-chip tag="span"
           data-attr="some attribute"
>
  <replace-with-grandchildren>
    <span data-attr="some attribute"
          class="c-bolt-chip"
    >
      <span class="c-bolt-chip__item-text">
        This is a chip
      </span>
    </span>
  </replace-with-grandchildren>
</bolt-chip>
`;

exports[`<bolt-chip> Component with link 1`] = `
<bolt-chip tag="a"
           url="!#"
>
  <replace-with-grandchildren>
    <a href="!#"
       class="c-bolt-chip"
    >
      <span class="c-bolt-chip__item-text">
        This is a chip with link
      </span>
    </a>
  </replace-with-grandchildren>
</bolt-chip>
`;
